skyportal/utils/gcn.py

```diff
@@ -266,46 +280,61 @@ def get_skymap_url(root, notice_type, timeout=10):
 
 
 def is_retraction(root):
-    retraction = root.find("./What/Param[@name='Retraction']")
-    if retraction is not None:
-        retraction = int(retraction.attrib['value'])
-        if retraction == 1:
-            return True
+    if isinstance(root, dict):
+        retraction = root.get('retraction')
+    else:
+        retraction = root.find("./What/Param[@name='Retraction']")
+        if retraction is not None:
+            retraction = int(retraction.attrib['value'])
+            if retraction == 1:
+                return True
+
     return False
 
 
 def get_skymap_cone(root):
     ra, dec, error = None, None, None
-    mission = urlparse(root.attrib['ivorn']).path.lstrip('/')
-    # Try error cone
-    loc = root.find('./WhereWhen/ObsDataLocation/ObservationLocation')
-    if loc is None:
-        return ra, dec, error
 
-    ra = loc.find('./AstroCoords/Position2D/Value2/C1')
-    dec = loc.find('./AstroCoords/Position2D/Value2/C2')
-    error = loc.find('./AstroCoords/Position2D/Error2Radius')
+    if isinstance(root, dict):
+        ra = root.get("ra")
+        dec = root.get("dec")
+        error = root.get("ra_dec_error")
+    else:
+        mission = urlparse(root.attrib['ivorn']).path.lstrip('/')
+        # Try error cone
+        loc = root.find('./WhereWhen/ObsDataLocation/ObservationLocation')
+        if loc is None:
+            return ra, dec, error
+
+        ra = loc.find('./AstroCoords/Position2D/Value2/C1')
+        dec = loc.find('./AstroCoords/Position2D/Value2/C2')
+        error = loc.find('./AstroCoords/Position2D/Error2Radius')
 
-    if None in (ra, dec, error):
-        return ra, dec, error
+        if None in (ra, dec, error):
```

Comment:
same comment for this method

Reply:
The function returns ra, dec, error at the end.